WordPress/Classes/ViewRelated/Reader/ReaderFollowedSitesViewController.swift

```diff
@@ -0,0 +1,332 @@
+
+
+    // MARK: - Setup
+
+
+    private func setupTableView() {
+        assert(tableViewController != nil, "The tableViewController must be assigned before configuring the tableView")
+
+        tableView = tableViewController.tableView
+        WPStyleGuide.resetReadableMarginsForTableView(tableView)
+
+        refreshControl = tableViewController.refreshControl!
+        refreshControl.addTarget(self, action: #selector(ReaderStreamViewController.handleRefresh(_:)), forControlEvents: .ValueChanged)
+    }
+
+
+    private func setupTableViewHandler() {
+        assert(tableView != nil, "A tableView must be assigned before configuring a handler")
+
+        tableViewHandler = WPTableViewHandler(tableView: tableView)
+        tableViewHandler.delegate = self
+    }
+
+
+    // MARK: - Configuration
+
+
+    func configureSearchBar() {
+        let placeholderText = NSLocalizedString("Enter a URL or a tag to follow", comment: "Placeholder text prompting the user to type the name of the URL or tag they would like to follow.")
+        let attributes = WPStyleGuide.defaultSearchBarTextAttributes(WPStyleGuide.grey())
+        let attributedPlaceholder = NSAttributedString(string: placeholderText, attributes: attributes)
+        UITextField.appearanceWhenContainedInInstancesOfClasses([UISearchBar.self, ReaderFollowedSitesViewController.self]).attributedPlaceholder = attributedPlaceholder
+        let textAttributes = WPStyleGuide.defaultSearchBarTextAttributes(WPStyleGuide.greyDarken30())
+        UITextField.appearanceWhenContainedInInstancesOfClasses([UISearchBar.self, ReaderFollowedSitesViewController.self]).defaultTextAttributes = textAttributes
+
+        searchBar.autocapitalizationType = .None
+        searchBar.translucent = false
+        searchBar.tintColor = WPStyleGuide.grey()
+        searchBar.barTintColor = WPStyleGuide.greyLighten30()
+        searchBar.backgroundImage = UIImage()
+        searchBar.returnKeyType = .Done
+        searchBar.setImage(UIImage(named: "icon-clear-textfield"), forSearchBarIcon: .Clear, state: .Normal)
+        searchBar.setImage(UIImage(named: "icon-reader-search-plus"), forSearchBarIcon: .Search, state: .Normal)
+    }
+
+
+    func configureNoResultsView() {
+        if let count = tableViewHandler.resultsController.fetchedObjects?.count where count > 0 {
+            noResultsView.removeFromSuperview()
+        } else {
+            view.addSubview(noResultsView)
+            noResultsView.centerInSuperview()
+        }
+    }
+
+
+    // MARK: - Instance Methods
+
+
+    private func syncSites() {
+        if isSyncing {
+            return
+        }
+        isSyncing = true
+        let service = ReaderTopicService(managedObjectContext: managedObjectContext())
+        service.fetchFollowedSitesWithSuccess({[weak self] in
+            self?.configureNoResultsView()
+            self?.refreshControl.endRefreshing()
+            self?.isSyncing = false
+
+            }) { [weak self] (error) in
```

Comment:
I think with success and failure callbacks, it's probably better not to use trailing closure syntax, so that both closures are named? (and you could drop the brackets around `error`)

```
service.fetchFollowedSitesWithSuccess({ [weak self] in
   ...
}, failure: { [weak self] error in
  ...
})
```